src/anomalib/data/csv.py

```diff
@@ -0,0 +1,302 @@
+
+    for csv_type, path in csvs.items():
+        filename, label = _prepare_files_labels_from_csv(path, csv_type, extensions)
+        filenames += filename
+        labels += label
+
+    samples = DataFrame({"image_path": filenames, "label": labels})
+    samples = samples.sort_values(by="image_path", ignore_index=True)
+
+    # Convert to absolute path if not
+    samples.image_path = samples.image_path.apply(lambda path: _resolve_path(path, root))
+
+    # Create label index for normal (0) and abnormal (1) images.
+    samples.loc[(samples.label == "normal") | (samples.label == "normal_test"), "label_index"] = 0
+    samples.loc[(samples.label == "abnormal"), "label_index"] = 1
+    samples.label_index = samples.label_index.astype("Int64")
+
+    # If a path to mask is provided, add it to the sample dataframe.
+
+    if mask_csv is not None and abnormal_csv is not None:
+        samples.loc[samples.label == "abnormal", "mask_path"] = samples.loc[
+            samples.label == "mask_dir"
+        ].image_path.values
+        samples["mask_path"].fillna("", inplace=True)
+        samples = samples.astype({"mask_path": "str"})
+
+        # make sure all every rgb image has a corresponding mask image.
+        assert (
+            samples.loc[samples.label_index == 1]
+            .apply(lambda x: Path(x.image_path).stem in Path(x.mask_path).stem, axis=1)
+            .all()
+        ), "Mismatch between anomalous images and mask images. Make sure the mask files \
+            folder follow the same naming convention as the anomalous images in the dataset \
+            (e.g. image: '000.png', mask: '000.png')."
+    else:
+        samples["mask_path"] = ""
+
+    # remove all the rows with temporal image samples that have already been assigned
+    samples = samples.loc[
+        (samples.label == "normal") | (samples.label == "abnormal") | (samples.label == "normal_test")
+    ]
+
+    # Ensure the pathlib objects are converted to str.
+    # This is because torch dataloader doesn't like pathlib.
+    samples = samples.astype({"image_path": "str"})
+
+    # Create train/test split.
+    # By default, all the normal samples are assigned as train.
+    #   and all the abnormal samples are test.
+    samples.loc[(samples.label == "normal"), "split"] = "train"
+    samples.loc[(samples.label == "abnormal") | (samples.label == "normal_test"), "split"] = "test"
+
+    # Get the data frame for the split.
+    if split:
+        samples = samples[samples.split == split]
+        samples = samples.reset_index(drop=True)
+
+    return samples
+
+
+class CSVDataset(AnomalibDataset):
+    """CSV dataset.
+    Args:
+        task (TaskType): Task type. (``classification``, ``detection`` or ``segmentation``).
+        transform (A.Compose): Albumentations Compose object describing the transforms that are applied to the inputs.
+        split (str | Split | None): Fixed subset split that follows from folder structure on file system.
```

Comment:
is this description accurate for the `CSV` data?

Reply:
If you're referring to the `split` args - they're still supported, will remove mention of folder structures.

Reply:
yeah, I mentioned folder structure stuff..